### PR TITLE
Fix group parameters' name with type Array

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #### Features
 
+* [#221](https://github.com/tim-vandecasteele/grape-swagger/pull/221): Fixed group parameters' name with type Array - [@u2](https://github.com/u2).
```
Comment (dblock) [inline CHANGELOG.md]: This is a fix, no? Should be in Fixes.

File: lib/grape-swagger.rb [hunk 2]
```diff
@@ -468,12 +468,28 @@ def setup(options)
...
+                      else
+                        new_key = k
+                        unless array_param.nil?
+                          if k.to_s.start_with?(array_param.to_s + '[')
+                            new_key = array_param.to_s + '[]' + k.to_s.split(array_param)[1]
+                          end
+                        end
+                        modified_route_params[new_key] = route.route_params[k]
+                      end
+                    end
```
Comment (dblock) [inline lib/grape-swagger.rb]: You should refactor this away out of here. Or maybe even better, why isn't this invoked from  `parse_params`?

**Comment by u2** [comment]
Yes,it's a fix, I will improve it.

**Comment by u2** [comment]
Updated.

**Comment by dblock** [comment]
Sorry I merged another PR that conflicts with this, would you mind rebasing? Apologies.

**Comment by u2** [comment]
It can be merged now, I think.

**Comment by dblock** [comment]
Merged via https://github.com/tim-vandecasteele/grape-swagger/commit/12dae157e9d1b633490c0aa44c809a61c4e9ac09. 

I squashed that other "merge and fix conflicts" commit by doing this:

```ruby
git checkout master
git checkout -b new-branch
git merge array_repair --squash
git commit ...
```

**Pull request closed without merging**